packages/core-data/src/entity-provider.js

```diff
@@ -221,6 +221,15 @@ export function useEntityBlockEditor( kind, name, { id: _id } = {} ) {
 			);
 
 			function updateAttributes( attributes ) {
+				// Only attempt to update attributes, if attributes is an object.
+				if (
+					! attributes ||
+					Array.isArray( attributes ) ||
+					typeof attributes !== 'object'
```

Comment:
I just had to double check for my own benefit: this could catch `null` but I think it'd be okay since `{ ...null }` returns `{}`

👍🏻

Reply:
Thanks for the extra checking!

The problem for `null` values is a little edge-casey, but an example would be if a block had an attribute that's a list of items where `null` is a valid value. For a somewhat contrived example, if we update the attributes in the test block's `block.json` to the following:

```
  "attributes": {
    "items": {
      "type": "array",
      "items": {
        "type": "string"
      },
      "default": [
        "apples",
        null,
        "oranges"
      ]
    }
  },
```

Then, if we remove the `! attributes` check in this `if` block so that we don't handle null / falsy values, then when `{ ...null }` turns the value into an empty object, the attributes are mutated.

Before / after:

<img width="296" alt="image" src="https://github.com/WordPress/gutenberg/assets/14988353/d548f1c8-6c27-459d-b61d-511bee25454d">

In practice, I think it's fairly unlikely that there'll be blocks out in the wild affected by it, but good to catch `null` values here and return as-is just in case. I reckon a good next step will be to add some extra tests like Ella mentioned so that we're covering these sorts of edge cases 🙂

Reply:
Thanks for taking the time to double check!